pkg/cloudprovider/cloudprovider.go

```diff
@@ -234,16 +238,26 @@ func (c *CloudProvider) GetSupportedNodeClasses() []status.Object {
 	return []status.Object{&v1.EC2NodeClass{}}
 }
 
-func getTags(ctx context.Context, nodeClass *v1.EC2NodeClass, nodeClaim *karpv1.NodeClaim) map[string]string {
+func getTags(ctx context.Context, nodeClass *v1.EC2NodeClass, nodeClaim *karpv1.NodeClaim) (map[string]string, error) {
+	var offendingTag string
+	if _, found := lo.Find(v1.RestrictedTagPatterns, func(exp *regexp.Regexp) bool {
+		for key := range nodeClass.Spec.Tags {
+			if exp.MatchString(key) {
+				offendingTag = key
```

Comment:
nit: If you loop through the `nodeClass.Spec.Tags` on the `Find` rather than looping through the `v1.RestrictedTagPatterns`, then you can just use the value that's returned from the find rather than having to do a separate value.